gems/aws-sdk-core/lib/aws-sdk-core/credentials_provider_chain.rb

```diff
@@ -2,91 +2,104 @@
 
 module AWS::SDK::Core
   # @api private
-  module CredentialsProviderChain
-    def self.call(config)
+  class CredentialsProviderChain < Hearth::IdentityProvider
+    def initialize(config)
```

Comment:
The way we're initializing this from the _defaults with the cfg provided by the resolver... this won't actually be the config, it will be the Config Resolver right?  Is there any risk or issues in keeping that around instead of using the resolved config?

Reply:
No, I think it should be fine. I do think after discussion that we want to cache providers based on config inputs, so that a new profile or some value that influences credential providers when given per operation, resolves a new provider in the chain.